### PR TITLE
MAINT: stats: raise error when input to kruskal has >1 dim

### LINKED ISSUE #9981: stats.kruskal : add a warning for an input with 2 or more columns
if a dataframe  has two columns, results are weird. For instance let the dataframe x be (first column = indices):
         counts  group
3803673      31      0
3800421      24      0
3800446      21      0
3804785      11      0
3801989      31      0
3801364      15      0
3804710      12      0
3800049      20      0
3801342      11      0
3804642       4      0

```stats.kruskal(x, x)```
gives this result: 
KruskalResult(statistic=205.08158702862053, pvalue=1.6254869278866505e-46)

I expect a warning message that the test is only suited for a series or a dataframe with one column.

### PR DESCRIPTION
**Pull request by mdhaber**
#### Reference issue
Closes gh-9981

#### What does this implement/fix?
`scipy.stats.kruskal` only works on 1D arrays (and scalars), but this isn't documented, and no error is raised if this condition is not met. This PR adds a check that the samples have no more than 1 non-singleton dimension

### CODE REVIEW

File: scipy/stats/stats.py [hunk 1]
```diff
@@ -7324,13 +7324,18 @@ def kruskal(*args, nan_policy='propagate'):
...
     args = list(map(np.asarray, args))
+
     num_groups = len(args)
     if num_groups < 2:
         raise ValueError("Need at least two groups in stats.kruskal()")
 
     for arg in args:
         if arg.size == 0:
             return KruskalResult(np.nan, np.nan)
+        elif np.squeeze(arg).ndim > 1:  # 0d (scalar) is ok
```
Comment (chrisb83) [inline scipy/stats/stats.py:7335]: in a few other tests (e.g. in _hypotests.py), an error is raised if `arg.ndim > 1`. I wonder if we should follow the same approach here, I did not understand the statement 'more than one non-singleton dimension' right away. it is a minor point, I am also fine with the current version.

Comment (mdhaber) [inline scipy/stats/stats.py:7335]:
```suggestion
        elif arg.ndim > 1:  # 0d (scalar) is ok
```

File: scipy/stats/stats.py [hunk 2]
```diff
@@ -7324,13 +7324,18 @@ def kruskal(*args, nan_policy='propagate'):
...
+            raise ValueError("Samples may not have more than one "
+                             "non-singleton dimension.")
```
Comment (mdhaber) [inline scipy/stats/stats.py:7337]:
```suggestion
            raise ValueError("Samples may not have more than one "
                             "dimension.")
```

File: scipy/stats/stats.py [hunk 3]
```diff
@@ -7271,7 +7271,7 @@ def kruskal(*args, nan_policy='propagate'):
     ----------
     sample1, sample2, ... : array_like
        Two or more arrays with the sample measurements can be given as
-       arguments.
+       arguments. Samples may not have more than one non-singleton dimension.
```
Comment (mdhaber) [inline scipy/stats/stats.py:7274]:
```suggestion
       arguments. Samples may not have more than one dimension.
```

File: scipy/stats/tests/test_stats.py [hunk 4]
```diff
@@ -5187,6 +5187,19 @@ def test_empty(self):
...
+        # N-d arrays are OK...
+        x = [[1]]
+        y = [[2]]  # this is fine
+        h, p = stats.kruskal(x, y)
+        assert_equal(h, 1.0)
+        assert_approx_equal(p, stats.distributions.chi2.sf(h, 1))
+
+        # but only one non-singleton dimension is allowed
+        z = np.random.rand(2, 2)
+        with assert_raises(ValueError, match="Samples may not have more"):
```
Comment (mdhaber) [inline scipy/stats/tests/test_stats.py:5200]:
```suggestion
        with assert_raises(ValueError, match="Samples must be one-dimensional."):
```

File: scipy/stats/tests/test_stats.py [hunk 5]
```diff
@@ -5187,6 +5187,19 @@ def test_empty(self):
...
         assert_equal(stats.kruskal(x, y, z), (np.nan, np.nan))
 
+    def test_nd_arrays(self):
+        # N-d arrays are OK...
+        x = [[1]]
+        y = [[2]]  # this is fine
+        h, p = stats.kruskal(x, y)
+        assert_equal(h, 1.0)
+        assert_approx_equal(p, stats.distributions.chi2.sf(h, 1))
+
```
Comment (mdhaber) [inline scipy/stats/tests/test_stats.py:5197]:
```suggestion
```

Comment (mdhaber) [inline scipy/stats/tests/test_stats.py:5197]:
```suggestion

```

File: scipy/stats/tests/test_stats.py [hunk 6]
```diff
@@ -5187,6 +5187,19 @@ def test_empty(self):
         z = []
         assert_equal(stats.kruskal(x, y, z), (np.nan, np.nan))
 
+    def test_nd_arrays(self):
+        # N-d arrays are OK...
+        x = [[1]]
+        y = [[2]]  # this is fine
```
Comment (mdhaber) [inline scipy/stats/tests/test_stats.py:5193]:
```suggestion
        x = [1]
        y = [2]
```

File: scipy/stats/tests/test_stats.py [hunk 7]
```diff
@@ -5187,6 +5187,19 @@ def test_empty(self):
         z = []
         assert_equal(stats.kruskal(x, y, z), (np.nan, np.nan))
 
+    def test_nd_arrays(self):
+        # N-d arrays are OK...
```
Comment (mdhaber) [inline scipy/stats/tests/test_stats.py:5191]:
```suggestion
        # Inputs must be exactly one-dimensional; check for error when input is not 1d
```

File: scipy/stats/tests/test_stats.py [hunk 8]
```diff
@@ -5187,6 +5187,19 @@ def test_empty(self):
...
+        x = [[1]]
+        y = [[2]]  # this is fine
+        h, p = stats.kruskal(x, y)
+        assert_equal(h, 1.0)
+        assert_approx_equal(p, stats.distributions.chi2.sf(h, 1))
+
+        # but only one non-singleton dimension is allowed
```
Comment (mdhaber) [inline scipy/stats/tests/test_stats.py:5198]:
```suggestion
```

**Review by WarrenWeckesser — APPROVED** [review]
Looks good, thanks @mdhaber.

**Pull request merged**